project-copacetic/copacetic
# feat: add ignore errors flags

- Added a new `--ignore-errors` flag
-  When this flag is used, even if we get an error while patching any of the packages, copa will still continue to produce an image.

Closes #203 

## How to test?

1. `trivy image --vuln-type os --ignore-unfixed -f json -o nginx.1.21.6.json mcr.microsoft.com/oss/nginx/nginx:1.21.6`
2.  First, run copa without the `--ignore-errors` flag:
```bash
sudo copa patch -i mcr.microsoft.com/oss/nginx/nginx:1.21.6 -r nginx.1.21.6.json -t 1.21.6-patched
```

![image](https://github.com/project-copacetic/copacetic/assets/43821031/bda56649-1f78-4be3-8302-6a190a1b1b47)
![image](https://github.com/project-copacetic/copacetic/assets/43821031/15b034ee-1c5d-4100-a923-ac2ba6d746d7)

3. Now run with the `--ignore-errors` flag enabled:
```bash
sudo copa patch -i mcr.microsoft.com/oss/nginx/nginx:1.21.6 -r nginx.1.21.6.json -t 1.21.6-patched --ignore-errors
```

![image](https://github.com/project-copacetic/copacetic/assets/43821031/a5512061-6525-43d7-9fd5-d06a682935d7)
![image](https://github.com/project-copacetic/copacetic/assets/43821031/c7a3768e-e4fe-4a9e-a8d3-57d45218fead)
![image](https://github.com/project-copacetic/copacetic/assets/43821031/f1bb7193-9f56-4747-99e0-08656ce11a93)

We can see that despite the 2 errors, copa ignores them, and still produces an image

4. Verified, that except those 2 packages, rest of the packages are patched:

![image](https://github.com/project-copacetic/copacetic/assets/43821031/67deed9c-de3a-4716-a278-c579f0a20887)
